### PR TITLE
Update fact_values_helper.rb

### PR DESCRIPTION
**Pull request by karl-ravn**
Corrected strange search for host_id parameter (breaks at least chef-facts integration)

### CODE REVIEW

**Comment by karl-ravn** [comment]
The link that broke has this when parameters.inspect is injected as "params":

```
<a href="/fact_values?parent_fact=ganglia"><i class="glyphicon glyphicon-plus-sign "></i> </a> <span><a href="/fact_values.%7B%22host_id%22=%3E%22labb-ix06.se-ix.delta.prod%22,%20%22locale%22=%3Enil,%20%22controller%22=%3E%22fact_values%22,%20%22action%22=%3E%22index%22,%20%22search%22=%3E%22host%20=%20labb-ix06.se-ix.delta.prod%22%7D" title="Show all ganglia children fact values">ganglia</a></span>
```

The problem was the searching, which is a rather confusing method, since the helper is only looking for one value in the params hash. Just fetching the value straight up is far easier.

**Comment by karl-ravn** [comment]
And now the link looks better also..

**Comment by domcleal** [comment]
@ares could you review please? [test]

**Comment by karl-ravn** [comment]
It doesn't work when going from the main menu.. only from a specific host..

**Comment by ares** [comment]
so the problem is with ruby 1.8 where select returns array instead of hash, I think more relevant fix would be to reconstruct hash from that array if it's the case, not sure what is the problem with links yet

**Comment by karl-ravn** [comment]
Now it works better and follows /hosts/<host_id> style when a host_id is used in the parameters, otherwise it goes to /fact_values

File: app/helpers/fact_values_helper.rb [hunk 1]
```diff
@@ -15,8 +15,14 @@ def fact_name(value, parent)
...
-          parameters = params.select { |k| k.to_s == 'host_id' }.merge(:parent_fact => memo)
-          link_to(current_name, fact_values_path(parameters),
+          parameters = {:parent_fact => memo} #params.select { |k,i| k.to_s == 'host_id' }.merge(:parent_fact => memo)
+          if (params[:host_id]) 
+            parameters.merge!({:host_id => params[:host_id]}) 
+            url=host_facts_path(parameters)
+          else
+            url=fact_values_path(parameters)
+          end
+          link_to(current_name, url,
```
Comment (ares) [inline app/helpers/fact_values_helper.rb:24]: I see, works nicely, I modified it to this (just nitpicks)

``` ruby
parameters = { :parent_fact => memo }
if params[:host_id]
  url = host_facts_path(parameters.merge({ :host_id => params[:host_id] }))
else
  url = fact_values_path(parameters)
end
```

basically removed () in if, put spaces in url=... and combined two lines, removed comment
could you please incorporate?

**Comment by dLobatog** [comment]
[test]

**Comment by ares** [comment]
nice, let's see whether [test] are green

**Comment by ares** [comment]
Seems like test errors were not caused by patch itself.

**Comment by domcleal** [comment]
Thanks @karl-ravn, merged as e72b0a7 for Foreman 1.4.1.

**Pull request closed without merging**